samples/python/speech_sample/speech_sample.py

```diff
@@ -83,24 +83,32 @@ def main():
             output_layer_names, output_layer_ports = parse_outputs_from_args(args)
             model.add_outputs(list(zip(output_layer_names, output_layer_ports)))
 
+        if args.layout:
+            layouts = parse_input_layouts(args, model.inputs)
+
         ppp = PrePostProcessor(model)
 
         for i in range(len(model.inputs)):
-            ppp.input(i).tensor() \
-                .set_element_type(Type.f32) \
-                .set_layout(Layout('NC'))  # noqa: N400
+            ppp.input(i).tensor().set_element_type(Type.f32)
+
+            input_name = model.input(i).get_any_name()
 
-            ppp.input(i).model().set_layout(Layout('NC'))
+            if args.layout and input_name in layouts.keys():
+                ppp.input(i).tensor().set_layout(Layout(layouts[input_name]))
+                ppp.input(i).model().set_layout(Layout(layouts[input_name]))
 
         for i in range(len(model.outputs)):
             ppp.output(i).tensor().set_element_type(Type.f32)
 
         model = ppp.build()
 
-        if args.context_window_left == args.context_window_right == 0:
-            set_batch(model, args.batch_size)
-        else:
-            set_batch(model, 1)
+        if args.batch_size:
+            batch_size = args.batch_size if args.context_window_left == args.context_window_right == 0 else 1
+
+            if any([not _input.node.layout.empty for _input in model.inputs]):
```

Comment:
I'm not sure that this check is correct, but it should work for most cases. 
@ilya-lavrenov , could you please take a look at this check, do we need to request the setting as a mandatory step to set the batch?